l1-contracts-foundry/script/DeployL1.s.sol

```diff
@@ -0,0 +1,736 @@
+            abi.encode(addresses.bridges.sharedBridgeImplementation, addresses.transparentProxyAdmin, initCalldata)
+        );
+        address contractAddress = deployViaCreate2(bytecode);
+        console.log("SharedBridgeProxy deployed at:", contractAddress);
+        addresses.bridges.sharedBridgeProxy = contractAddress;
+    }
+
+    function registerSharedBridge() internal {
+        Bridgehub bridgehub = Bridgehub(addresses.bridgehub.bridgehubProxy);
+        vm.startBroadcast();
+        bridgehub.addToken(ADDRESS_ONE);
+        bridgehub.setSharedBridge(addresses.bridges.sharedBridgeProxy);
+        vm.stopBroadcast();
+        console.log("SharedBridge registered");
+    }
+
+    function deployErc20BridgeImplementation() internal {
+        bytes memory bytecode = abi.encodePacked(
+            type(L1ERC20Bridge).creationCode,
+            abi.encode(addresses.bridges.sharedBridgeProxy)
+        );
+        address contractAddress = deployViaCreate2(bytecode);
+        console.log("Erc20BridgeImplementation deployed at:", contractAddress);
+        addresses.bridges.erc20BridgeImplementation = contractAddress;
+    }
+
+    function upgradeL1Erc20Bridge() internal {
+        // In local network, we need to change the block.number
+        // as the operation could be scheduled for timestamp 1
+        // which is also a magic number meaning the operation
+        // is done.
+        if (config.l1ChainId == 31337) {
+            vm.warp(10);
+        }
+
+        bytes memory callData = abi.encodeCall(
+            ProxyAdmin.upgradeAndCall,
+            (
+                ITransparentUpgradeableProxy(addresses.bridges.erc20BridgeProxy),
+                addresses.bridges.erc20BridgeImplementation,
+                abi.encodeCall(L1ERC20Bridge.initialize, ())
+            )
+        );
+
+        IGovernance.Call[] memory calls = new IGovernance.Call[](1);
+        calls[0] = IGovernance.Call({target: addresses.transparentProxyAdmin, value: 0, data: callData});
+
+        IGovernance.Operation memory operation = IGovernance.Operation({
+            calls: calls,
+            predecessor: bytes32(0),
+            salt: bytes32(0)
+        });
+
+        Governance governance = Governance(payable(addresses.governance));
+
+        vm.startBroadcast();
+        governance.scheduleTransparent(operation, 0);
+        governance.execute(operation);
+        vm.stopBroadcast();
+        console.log("L1Erc20Bridge upgraded");
+    }
+
+    function updateOwners() internal {
+        vm.startBroadcast();
+
+        ValidatorTimelock validatorTimelock = ValidatorTimelock(addresses.validatorTimelock);
+        validatorTimelock.transferOwnership(config.ownerAddress);
+
+        Governance governance = Governance(payable(addresses.governance));
+        governance.transferOwnership(config.ownerAddress);
+
+        addresses.bridgehub.bridgehubProxy.call(abi.encodeCall(Ownable2Step.transferOwnership, (config.ownerAddress)));
```

Comment:
why low-level call?

Reply:
`transferOwnership` is not visible in the proxy, I needed to use a low-level call to go through the proxy.

![image](https://github.com/matter-labs/era-contracts/assets/21188659/6d177dae-4bdd-4135-b983-741c6dfe2d9f)

Reply:
Fixed